util/network-alert/src/notifier.rs

```diff
@@ -0,0 +1,93 @@
+use ckb_core::alert::Alert;
+use ckb_logger::warn;
+use fnv::FnvHashMap;
+use lru_cache::LruCache;
+use std::sync::Arc;
+
+const CANCEL_FILTER_SIZE: usize = 128;
+
+pub struct Notifier {
+    /// cancelled alerts
+    cancel_filter: LruCache<u32, ()>,
+    /// alerts we received
+    received_alerts: FnvHashMap<u32, Arc<Alert>>,
```

Comment:
are `received_alerts` values same as `noticed_alerts`? why need to store duplicate data here?

Reply:
https://github.com/nervosnetwork/ckb/pull/910#discussion_r289673881